src/Microsoft.AspNetCore.Razor.Language/MemoryRazorProject.cs

```diff
@@ -0,0 +1,201 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.AspNetCore.Razor.Language
+{
+    public class MemoryRazorProject : RazorProject
+    {
+        private readonly DirectoryNode RootNode = new DirectoryNode("/", "/");
```

Comment:
I'd yank this guy out of this class and require the `VirtualProjectFileSystem` to be constructed with one. Also would rename it to `VirtualFileSystemHierarchy`